src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/TelemetryService.kt

```diff
@@ -18,6 +20,10 @@ class TelemetryService(
 
   fun trackGoalCreateEvent(goal: Goal) {
     telemetryClient.trackEvent(GOAL_CREATE_EVENT, goal.goalCreateEventCustomDimensions())
+    with(Span.current()) {
```

Comment:
This is the new code (thats not covered by unit tests at the moment) - its a hunch that we might be able to send events using the underlying `opentelemetry` API (the `Span` class) , in which case it might use all the custom dimensions that were already set on the `Span` via the interceptor.
At the moment I am sending both events - one via the `TelemetryClient` class which we know works (albeit without the custom dimensions), and this way as well. I've given this approach (using the `Span`) a subtly different event name (for now) so we can identify it in the app insights query tool